google-services-plugin/src/main/groovy/com/google/gms/googleservices/GoogleServicesTask.java

```diff
@@ -74,8 +75,8 @@
 
   private File intermediateDir;
   private String variantDir;
-  private String packageNameXOR1;
-  private TextResource packageNameXOR2;
+  private TextResource packageName;
```

Comment:
don't use a `TextResource`. Use a `Property<String>`. You can declare it as 

```
@Input
public Property<String> getPackageName()
```
(though you should really rename this `applicationId`

Then you can set it in the plugin with
`task.applicationId.set(variant.applicationId)`